storage/innobase/log/log0online.cc

```diff
@@ -934,11 +934,8 @@ void log_online_read_init(void) {
         bitmap_dir->dir_entry[i].mystat->st_size > 0) {
       log_bmp_sys->out_seq_num = file_seq_num;
       last_file_start_lsn = file_start_lsn;
-      /* No dir component (log_bmp_sys->bmp_file_home) here, because
-that's the cwd */
-      strncpy(log_bmp_sys->out.name, bitmap_dir->dir_entry[i].name,
-              FN_REFLEN - 1);
-      log_bmp_sys->out.name[FN_REFLEN - 1] = '\0';
+      snprintf(log_bmp_sys->out.name, sizeof(log_bmp_sys->out.name), "%s%s",
+               log_bmp_sys->bmp_file_home, bitmap_dir->dir_entry[i].name);
```

Comment:
are you sure the bmp_file_home always contains the path separator at end?

Reply:
It's initialized in the following way, so it should always be there

`  /* Initialize bitmap file directory from srv_data_home and add a path
  separator if needed.  */
  const size_t srv_data_home_len = strlen(srv_data_home);
  ut_a(srv_data_home_len < FN_REFLEN);
  strcpy(log_bmp_sys->bmp_file_home, srv_data_home);
  if (srv_data_home_len &&
      log_bmp_sys->bmp_file_home[srv_data_home_len - 1] != SRV_PATH_SEPARATOR) {
    ut_a(srv_data_home_len < FN_REFLEN - 1);
    log_bmp_sys->bmp_file_home[srv_data_home_len] = SRV_PATH_SEPARATOR;
    log_bmp_sys->bmp_file_home[srv_data_home_len + 1] = '\0';
  }
`